apps/aecore/lib/aecore/chain/chain_state.ex

```diff
@@ -5,120 +5,129 @@ defmodule Aecore.Chain.ChainState do
   """
 
   alias Aecore.Structures.SignedTx
+  alias Aecore.Structures.SpendTx
+  alias Aecore.Structures.DataTx
   alias Aecore.Structures.Account
   alias Aeutil.Serialization
   alias Aeutil.Bits
 
   require Logger
 
-  @type account_chainstate() :: %{binary() => map()}
+  @typedoc "Public key representing an account"
+  @type pubkey() :: binary()
 
-  @spec calculate_and_validate_chain_state!(list(), account_chainstate(), integer()) :: account_chainstate()
-  def calculate_and_validate_chain_state!(txs, chain_state, block_height) do
+  @typedoc "State of an account"
+  @type acc_state() :: %{balance: integer(),
+                         locked: [%{amount: integer(), block: integer()}],
+                         nonce: integer()}
+
+  @typedoc "Structure of the accounts"
+  @type accounts() :: %{pubkey() => acc_state()}
+
+  @typedoc "Structure of the chainstate"
+  @type chainstate() :: map()
```

Comment:
this can be more specific than `map()` e.g. 
`@type chain_state_types() :: :accounts`
`@type chainstate() :: %{chain_state_types() => accounts()}`
(not sure if correct example)